test/date_types.test.js

```diff
@@ -232,7 +232,7 @@ describe('MySQL DATE, DATETTIME, TIMESTAMP types on server with non local TZ (+0
 });
 
 var prepareModel = function(tz, done) {
-  db = getSchema({timezone: tz});
+  db = getSchema({timezone: tz, legacyDateTypeProcessing:false});
```

Comment:
Why are we defaulting to false? We should be defaulting to true to preserve old behaviour and forcing users to turn on the flag for the new major/breaking behaviour.